src/components/Filter/hooks/useMultiFilterUtils.ts

```diff
@@ -0,0 +1,114 @@
+import { isEqual } from 'lodash';
+import { useEffect, useState } from 'react';
+
+import { FilterOption } from '../types';
+import { Props as FilterProps } from '../types';
+import { SELECT_ALL_OPTION } from 'components/Select/constants';
+
+type Props = Pick<
+  FilterProps,
+  'multi' | 'selectedItems' | 'items' | 'defaultValue' | 'onClear' | 'onFilterDelete'
+> & {
+  setFilterLabel: (value: string) => void;
+  setIsOpen: (value: boolean) => void;
+  setSearchValue: (value: string) => void;
+};
+
+const useMultiFilterUtils = ({
+  multi,
+  selectedItems = [],
+  items,
+  defaultValue,
+  setFilterLabel,
+  setIsOpen,
+  onClear,
+  setSearchValue,
+  onFilterDelete = () => {},
+}: Props) => {
+  const [availableMultiFilters, setAvailableMultiFilters] = useState<FilterOption[]>([]);
+
+  const [multiFilters, setMultiFilters] = useState(selectedItems);
+
+  useEffect(() => {
+    if (multi) {
```

Comment:
I get the way this hook is used, but by naming it `useMultiFilterUtils` you'd expect that whenever this hook runs, `multi` is `true`. Maybe this needs to be renamed, as it is used in the Filter.tsx and not in the MultiFilter component, or change the way it's used? Thoughts?

Reply:
this hook is created to contain all the necessary callbacks and states for the case where `multi=true`. Therefore it's named `useMultiFilterUtils`. We need these functions and states on the Filter level (and not MultiFilter) so it cannot be called anywhere else. It was created to reduce the size of the `Filter` component. It follows the pattern of `useMultiSelectUtils` in Select component